x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java

```diff
@@ -152,7 +130,7 @@ protected boolean initialRun() {
     @Override
     protected void onFinish(ActionListener<Void> listener) {
         // reset the page size, so we do not memorize a low page size forever, the pagesize will be re-calculated on start
-        pageSize = 0;
+        pageSize = pivot.getInitialPageSize();
```

Comment:
I find it more logical to re-calc it at start (line 118). It doesn't matter right now but maybe in future the page size calculation depends on a setting or entails some more other data dependent calculation.

Reply:
so the comment "the pagesize will be re-calculated on start" is not true anymore after this change